src/vantage6_strongaya_general/miscellaneous.py

```diff
@@ -0,0 +1,665 @@
+                'education': ['primary', 'secondary'],
+                'age': {'start': 15, 'end': 39},
+                'height': {'end': 196}
+            }
+
+    Returns:
+        pd.DataFrame: The stratified DataFrame.
+    """
+    if (
+        variables_to_stratify is None
+        or not isinstance(variables_to_stratify, dict)
+        or len(variables_to_stratify) == 0
+    ):
+        return df
+
+    safe_log("info", "Stratifying data based on specified variables")
+
+    # Check if the specified variables exist in the DataFrame
+    for variable in variables_to_stratify:
+        if variable not in df.columns:
+            safe_log(
+                "error",
+                f"Variable '{variable}' not found in DataFrame. Data stratification will not be applied",
+            )
+            return df
+
+    # Build the query string
+    query_conditions = []
+    for variable, values in variables_to_stratify.items():
+        if isinstance(values, dict):
+            # Handle range dictionary
+            start = values.get("start")
+            end = values.get("end")
+            if start is not None and end is not None:
+                query_conditions.append(
+                    f"`{variable}` >= {start} and `{variable}` <= {end}"
+                )
+            elif start is not None:
+                query_conditions.append(f"`{variable}` >= {start}")
+            elif end is not None:
+                query_conditions.append(f"`{variable}` <= {end}")
+        else:
+            # Handle the list of values
+            query_conditions.append(f"`{variable}` in {values}")
+
+    query_string = " and ".join(query_conditions)
+
+    # Apply the query to filter the DataFrame
+    stratified_df = df.query(query_string)
+
+    return stratified_df.reset_index(drop=True)
+
+
+def collect_organisation_ids(
+    organisation_ids: Optional[List[int]], client: AlgorithmClient
+) -> List[int]:
+    """
+    Collect organisation IDs, ensuring they are a list of integers.
+    If the input is not a list of integers, attempt to convert it.
+    If the input is None, collect all organisation IDs.
+
+    Args:
+        organisation_ids (Optional[List[int]]): List of organisation IDs.
+        client (AlgorithmClient): The client object to interact with the organisation API.
+
+    Returns:
+        List[int]: A list of valid organisation IDs.
+    """
+    if organisation_ids is None:
+        safe_log(
+            "info", "No organisation IDs provided. Collecting all organisation IDs"
+        )
+        organisations = client.organization.list()
+        return [organisation.get("id") for organisation in organisations]
+
+    if isinstance(organisation_ids, list):
+        try:
+            # Attempt to convert all elements to integers
+            organisation_ids = [int(i) for i in organisation_ids]
+        except ValueError:
+            safe_log("error", "Organisation IDs should be a list of integers")
+            return []
```

Comment:
Why return an empty list here? Won't that lead to issues later when the central function tries to create subtasks for 0 orgs and then gets an error? Maybe you can raise a `UserInputError` here - see https://github.com/vantage6/vantage6/blob/main/vantage6-algorithm-tools/vantage6/algorithm/tools/exceptions.py